src/main/java/org/reaktivity/nukleus/http2/internal/Http2WriteScheduler.java

```diff
@@ -276,18 +281,19 @@ public boolean data(int streamId, DirectBuffer buffer, int offset, int length)
         }
 
 
-        if (!buffered() && !buffered(streamId) && sizeof <= connection.outWindow && length <= connection.http2OutWindow &&
+        if (!buffered() && !buffered(streamId) && hasNukleusWindowBudget(length) && length <= connection.http2OutWindow &&
                 length <= stream.http2OutWindow)
         {
             // Send multiple DATA frames (because of max frame size)
             while (length > 0)
             {
                 int chunk = Math.min(length, connection.remoteSettings.maxFrameSize);
                 Flyweight.Builder.Visitor data = http2Writer.visitData(streamId, buffer, offset, chunk);
-                http2(stream, type, chunk + 9, data);
+                http2(stream, type, chunk + 9, data, false);
```

Comment:
Please extract `9` into a named constant to make it clearer how it was computed, and then reference it accordingly throughout the code.